xbmc/cores/VideoPlayer/VideoRenderers/LinuxRendererGLES.cpp

```diff
@@ -279,24 +279,29 @@ void CLinuxRendererGLES::LoadPlane(CYuvPlane& plane, int type,
 
   glBindTexture(m_textureTarget, plane.id);
 
-  if (m_pixelStoreKey > 0)
+  GLint pixelStore = -1;
+  if (stride != static_cast<int>(width * bps))
   {
-    glPixelStorei(m_pixelStoreKey, stride);
-  }
-  else if (stride != static_cast<int>(width * bps))
-  {
-    unsigned char *src(static_cast<unsigned char*>(data)),
-                  *dst(m_planeBuffer);
+    if (m_pixelStoreKey > 0)
+    {
+      glGetIntegerv(m_pixelStoreKey, &pixelStore);
```

Comment:
You really shouldn't call glGet functions in typical usage. It will likely cause gl rendering pipeline to flush.
https://www.khronos.org/opengl/wiki/Common_Mistakes

> glGetFloatv glGetBooleanv glGetDoublev glGetIntegerv
You find that these functions are slow.
That's normal. Any function of the glGet form will likely be slow. nVidia and ATI/AMD recommend that you avoid them. The GL driver (and also the GPU) prefer to receive information in the up direction. You can avoid all glGet calls if you track the information yourself.

Ideally keep track of these through a local variable, not by asking the GPU.